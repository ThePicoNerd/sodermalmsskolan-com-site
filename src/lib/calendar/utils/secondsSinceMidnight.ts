/**
 * Get the number of seconds since midnight for a particular date.
 *
 * @param {Date} date Input date.
 *
 * @returns {number} The number of seconds.
 */
const secondsSinceMidnight = (date: Date): number => date.getHours() * 3600
 + date.getMinutes() * 60
 + date.getSeconds();

export default secondsSinceMidnight;
