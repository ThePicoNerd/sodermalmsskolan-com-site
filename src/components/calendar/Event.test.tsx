import { cleanup, render } from "@testing-library/react";
import dayjs from "dayjs";
import React from "react";
import secondsSinceMidnight from "../../lib/calendar/utils/secondsSinceMidnight";
import CalendarEventView, { CalendarEventViewProps } from "./Event";

const startDate = dayjs("2020-01-01 8:10");
const start = secondsSinceMidnight(startDate.toDate());

type DeepPartial<T> = {
  [P in keyof T]?: DeepPartial<T[P]>;
};

/**
 * Use a pre-filled object to get props.
 *
 * @param {DeepPartial<CalendarEventViewProps>} partialProps Partial props.
 *
 * @returns {CalendarEventViewProps} The complete props.
 */
const getEventProps = (
  partialProps: DeepPartial<CalendarEventViewProps> = {},
): CalendarEventViewProps => ({
  start,
  data: {
    duration: 3900,
    title: "My event",
    deltaStart: 0,
    deltaDuration: 0,
    characteristics: ["1"],
    signature: "1",
    ...partialProps?.data,
  },
});

describe("<CalendarEventView />", () => {
  afterEach(cleanup);

  it("renders properly", () => {
    const { container } = render(<CalendarEventView {...getEventProps()} />);

    expect(container).toMatchSnapshot();
  });

  it("renders properly as a placeholder", () => {
    const { container } = render(
      <CalendarEventView
        {...getEventProps({
          data: {
            placeholder: true,
          },
        })}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
