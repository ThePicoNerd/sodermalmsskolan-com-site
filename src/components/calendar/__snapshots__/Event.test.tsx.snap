// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarEventView /> renders properly 1`] = `
.emotion-0 {
  width: 100%;
  height: 6rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  list-style: none;
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    position: absolute;
    top: calc(((29400 / var(--row-duration)) - var(--row-pad-start)) * var(--row-height));
    height: calc((3900 / var(--row-duration)) * var(--row-height));
    width: 100%;
    left: 0%;
    margin: 0;
  }
}

.emotion-1 {
  background-color: #000;
  border-radius: 0.3125rem;
  border: 0;
  border-left: 4px solid #000;
  box-sizing: border-box;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  padding: 0.625rem;
  color: #ffffff;
  height: 100%;
  position: relative;
  -webkit-transition: opacity 0.1s;
  transition: opacity 0.1s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font: inherit;
  text-align: left;
  width: 100%;
  outline: none;
}

@media (prefers-color-scheme: dark) {
  .emotion-1 {
    background-color: rgba(0,0,0,0.5);
    border-color: #000;
    color: #ffffff;
  }
}

.emotion-1:hover {
  z-index: 1;
}

.emotion-2 {
  font-family: IBM Plex Mono,monospace;
  font-size: 0.8125rem;
  font-weight: 400;
  opacity: 0.7;
  line-height: 1;
  margin-bottom: 0.5rem;
  display: inline-block;
}

.emotion-2>*:not(:first-of-type)::before {
  content: " · ";
}

.emotion-6 {
  font-size: 1rem;
  font-weight: 500;
}

.emotion-7 img {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}

<div>
  <li
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <span
          class="emotion-3"
        >
          <time>
            8:10
            <span
              class="emotion-4"
            >
              –
              9:15
            </span>
          </time>
        </span>
        <span
          class="emotion-3"
        />
      </div>
      <div
        class="emotion-6"
      >
        <span
          class="emotion-7"
        >
          My event
        </span>
      </div>
    </div>
  </li>
</div>
`;

exports[`<CalendarEventView /> renders properly as a placeholder 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-background-position: 200% 0;
    background-position: 200% 0;
  }

  100% {
    -webkit-background-position: -200% 0;
    background-position: -200% 0;
  }
}

.emotion-0 {
  width: 100%;
  height: 6rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  list-style: none;
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    position: absolute;
    top: calc(((29400 / var(--row-duration)) - var(--row-pad-start)) * var(--row-height));
    height: calc((3900 / var(--row-duration)) * var(--row-height));
    width: 100%;
    left: 0%;
    margin: 0;
  }
}

.emotion-1 {
  background-color: var(--skeleton-base);
  background-image: linear-gradient(270deg, var(--skeleton-base), var(--skeleton-highlight), var(--skeleton-base));
  background-repeat: no-repeat;
  -webkit-background-size: 200% 100%;
  background-size: 200% 100%;
  -webkit-animation: animation-0 2s ease-in-out infinite;
  animation: animation-0 2s ease-in-out infinite;
  border-radius: 0.3125rem;
  display: inline-block;
  line-height: 1;
  width: 100%;
  height: 100%;
}

<div>
  <li
    class="emotion-0"
  >
    <span
      class="emotion-1"
    />
  </li>
</div>
`;
